### PR TITLE
fix: circular dependency

### PR DESCRIPTION
**Pull request by TalDerei**
References https://github.com/penumbra-zone/penumbra/issues/3389. This PR migrates the benches from `proof-params` to a dedicated `bench` package and removes any unnecessary dev-dependencies to bypass the circular dependency bug that breaks rust-analyzer.

### CODE REVIEW

**Comment by conorsch** [comment]
CI looks happy, which is a great sign. If I run `cargo bench` locally, however, I see a bunch of warnings:

```
❯ cargo bench
warning: invalid feature `proving-keys` in required-features of target `nullifier_derivation`: `proving-keys` is not present in [features] section
warning: invalid feature `proving-keys` in required-features of target `output`: `proving-keys` is not present in [features] section
warning: invalid feature `proving-keys` in required-features of target `delegator_vote`: `proving-keys` is not present in [features] section
warning: invalid feature `proving-keys` in required-features of target `spend`: `proving-keys` is not present in [features] section
warning: invalid feature `proving-keys` in required-features of target `swap`: `proving-keys` is not present in [features] section
warning: invalid feature `proving-keys` in required-features of target `swap_claim`: `proving-keys` is not present in [features] section
warning: invalid feature `proving-keys` in required-features of target `undelegate_claim`: `proving-keys` is not present in [features] section
   Compiling penumbra-custody v0.63.1 (/home/conor/src/penumbra/crates/custody)
   Compiling penumbra-view v0.63.1 (/home/conor/src/penumbra/crates/view)
   Compiling narsil v0.63.1 (/home/conor/src/penumbra/crates/narsil/narsil)
   Compiling pd v0.63.1 (/home/conor/src/penumbra/crates/bin/pd)
[...snip...]
```
I made sure to confirm that these warnings do not appear on `main`. Anything we can do to clean them up? Once we're happy with the substantive changes, please make sure to edit the docs to make sure the correct directories and commands are specified, e.g. at https://guide.penumbra.zone/main/dev/parameter_setup.html

**Comment by cronokirby** [comment]
I don't want to bikeshed too hard either, but conceptually it is sort of weird to have these things in proof-setup, which is basically just the internal math necessary for the summoning ceremony. The reason that crate has a dependency on the circuits is just to calculate their size.

**Comment by TalDerei** [comment]
@conorsch I just noticed this as well. Per #3337, it requires passing in the 

```
penumbra-proof-params = { path = "../../crypto/proof-params", features = [
    "bundled-proving-keys",
    "download-proving-keys",
] }
```

dependency to the Cargo file. I'll incorporate the change. 

@cronokirby I agree. I sort of put the benches there as a placeholder to satisfy the CI, but moving it should be simple. I'll make that change as well.

**Comment by TalDerei** [comment]
I had to merge the commits from https://github.com/penumbra-zone/penumbra/issues/3389 to pass the CI and prevent the changes in this branch from breaking once merged into master. Keeping this PR as a draft until the other PR is merged.

**Comment by TalDerei** [comment]
Ready for review!

File: crates/crypto/proof-params/src/lib.rs [hunk 1]
```diff
@@ -78,8 +78,8 @@ impl Deref for LazyProvingKey {
     }
 }
 
-// Note: Conditionally load the proving key objects if the
-// bundled-proving-keys feature is enabled.
+/// Note: Conditionally load the proving key objects if the
+/// bundled-proving-keys is present.
```
Comment (hdevalence) [inline crates/crypto/proof-params/src/lib.rs:82]: I think this slipped in during rebasing — making these doc commands will prepend them onto the doc string of the next item, which isn't what we want.

**Review by cronokirby — APPROVED** [review]
I like the approach here; let's rebase and then get this merged

File: crates/crypto/proof-params/src/lib.rs (hunk 1 above)
Comment (TalDerei) [reply crates/crypto/proof-params/src/lib.rs]: changed

File: docs/guide/src/dev/parameter_setup.md [hunk 2]
```diff
@@ -33,10 +33,10 @@ Then edit `penumbra-proof-params` to reference the new parameters created in
 
 ## Circuit Benchmarks
 
-We have benchmarks for all proofs in the `penumbra-proof-params` crate. You can run them via:
+We have benchmarks for all proofs in the `penumbra-bench` crate. You can run them via:
 
 ```shell
-cd crates/crypto/proof-params
+cd crates/bench
```
Comment (conorsch) [inline docs/guide/src/dev/parameter_setup.md:39]: Nit: do we really need to `cd`? Running `cargo bench` at the top level repo root seems to work for me.

**Review by conorsch — CHANGES_REQUESTED** [review]
One more thing: when I run `cargo bench` on this branch, it updates the workspace's `Cargo.lock`. The changes to that file should be committed and added to this PR. Other than that, LGTM!

File: docs/guide/src/dev/parameter_setup.md (hunk 2 above)
Comment (TalDerei) [reply docs/guide/src/dev/parameter_setup.md:39]: we don't, removed

**Review by conorsch — APPROVED** [review]

**Comment by conorsch** [comment]
I'm going to do a bit of housekeeping on the commit history here, then merge.

**Pull request merged**